src/astro/databases/snowflake.py

```diff
@@ -1,17 +1,135 @@
 """Snowflake database implementation."""
-from typing import Dict, List, Tuple
+import logging
+import random
+import string
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Tuple
 
 import pandas as pd
 from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
 from pandas.io.sql import SQLDatabase
 from snowflake.connector import pandas_tools
-
-from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from snowflake.connector.errors import ProgrammingError
+
+from astro.constants import (
+    DEFAULT_CHUNK_SIZE,
+    FileLocation,
+    FileType,
+    LoadExistStrategy,
+    MergeConflictStrategy,
+)
 from astro.databases.base import BaseDatabase
+from astro.files import File
 from astro.sql.table import Metadata, Table
 
 DEFAULT_CONN_ID = SnowflakeHook.default_conn_name
 
+ASTRO_SDK_TO_SNOWFLAKE_FILE_FORMAT_MAP = {
+    FileType.CSV: "CSV",
+    FileType.NDJSON: "JSON",
+    FileType.PARQUET: "PARQUET",
+}
+
+COPY_OPTIONS = {
+    FileType.CSV: "ON_ERROR=CONTINUE",
```

Comment:
I'm worried about us using `ON_ERROR=CONTINUE` and hiding exceptions from users, giving the wrong sense of feeling that an operation was successful while many (or even all) rows may not have been successfully loaded.

I'll log a ticket for us to look into using/exposing the `VALIDATION_MODE` option, which could allow us to give more visibility to end-users.